package.json

```diff
@@ -225,47 +236,68 @@
         }
     },
     "scripts": {
-        "precommit": "pretty-quick --staged",
+        "format": "prettier --loglevel error --write \"{src,test}/**/*.ts\"",
+        "postformat": "npm run lint -- --fix",
         "preinstall": "git submodule update --init --recursive",
         "postinstall": "vscode-install",
-        "prebuild": "npm run lint && npm run test",
-        "build": "node ./out/src/icon-manifest/build.js",
-        "pretest": "rimraf ./.nyc_output ./coverage && npm run compile",
+        "prebuild": "npm run compile",
+        "build": "node ./out/src/build.js",
+        "prebuild:dev": "npm run lint && npm run compile:dev",
+        "build:dev": "node ./out/src/build.js",
+        "pretest": "rimraf ./.nyc_output ./coverage && npm run lint && npm run compile:dev",
         "test": "nyc -x '' mocha",
         "posttest": "nyc report -r lcov",
         "test:vs": "node ./node_modules/vscode/bin/test",
         "vscode:prepublish": "npm run build",
         "vscode:uninstall": "node ./out/src/uninstall.js",
         "precompile": "rimraf ./out",
-        "compile": "tsc",
-        "compile:w": "npm run compile -- -w",
+        "compile": "tsc -p tsconfig.prod.json",
+        "compile:w": "npm run compile:dev -- -w",
+        "precompile:dev": "npm run precompile",
+        "compile:dev": "tsc",
         "lint": "tslint -p .",
         "example": "example"
     },
     "devDependencies": {
-        "@types/chai": "^4.1.3",
-        "@types/lodash": "^4.14.109",
-        "@types/mocha": "^5.2.2",
-        "@types/mockery": "^1.4.29",
-        "@types/node": "7",
+        "@types/chai": "^4.1.7",
+        "@types/lodash": "^4.14.118",
+        "@types/mocha": "^5.2.5",
+        "@types/node": "8",
+        "@types/opn": "^5.1.0",
+        "@types/proxyquire": "^1.3.28",
         "@types/semver": "^5.5.0",
-        "@types/sinon": "^5.0.1",
-        "chai": "^4.1.2",
+        "@types/sinon": "^5.0.5",
+        "chai": "^4.2.0",
         "examples-generator": "file:submodules/examples-generator",
+        "husky": "^1.2.0",
+        "lint-staged": "^8.1.0",
         "mocha": "^5.2.0",
-        "mockery": "^2.1.0",
-        "nyc": "^12.0.2",
-        "prettier": "^1.14.0",
-        "pretty-quick": "^1.6.0",
+        "nyc": "^13.1.0",
+        "prettier": "^1.15.3",
+        "proxyquire": "^2.1.0",
         "rimraf": "^2.6.2",
-        "sinon": "^6.0.0",
+        "sinon": "^7.1.1",
         "tslint": "^5.11.0",
-        "typescript": "^2.9.1",
-        "vscode": "^1.1.18"
+        "typescript": "^3.2.2",
+        "vscode": "^1.1.26"
```

Comment:
Revert to `1.1.18`

Reply:
The thing is that `vscode@1.1.18` had a dependency with the wrong `event-stream` package which has been removed from npm and it was breaking. That's why I removed the tests. Is it important to stick to this version?

Reply:
Let me check again to see if I can sort it.